src/controllers/api.ts

```diff
@@ -78,13 +79,21 @@ export const postForceJoinOrganizationController = async (
       query: req.query,
     });
 
-    const userOrganizationLink = await forceJoinOrganization({
+    let userOrganizationLink = await getUserOrganizationLink(
       organization_id,
-      user_id,
-      is_external,
-    });
+      user_id
+    );
+    if (!userOrganizationLink) {
+      userOrganizationLink = await forceJoinOrganization({
+        organization_id,
+        user_id,
+        is_external,
+      });
+    }
 
-    await notifyAllMembers(userOrganizationLink);
+    if (!userOrganizationLink.authentication_by_peers_type) {
```

Comment:
Just to be sure, you're only notify all members when there is no authentication_by_peers_type? It is not the opposite = you need to notify all the members when there is an authentication_by_peers_type.